### PR TITLE
Visitor can read specific article

### PR DESCRIPTION
**Pull request by jysmys**
[PT link](https://www.pivotaltracker.com/story/show/173103470)
```
As a Visitor
In order to read an article
I would like to be able to click on an article and have it displayed
```
- added packages react-navigation with stack, web and native
- added react-native-gesture-handler

image

### CODE REVIEW

**Review by designerofthing — APPROVED** [review]

File: src/components/ArticleList.js [hunk 1]
```diff
@@ -3,9 +3,8 @@ import axios from "axios";
 import ArticleCard from "../components/ArticleCard";
 import { FlatList, View } from "react-native";
 
-const ArticleList = () => {
-  const [articleList, setArticleList] = useState([]);
-
+const ArticleList = ({ navigation }) => {
+  const [articleList, setArticleList] = useState(false);
```
Comment (erikbjoern) [inline src/components/ArticleList.js]: Should it be false? Why not empty array?

File: src/components/SingleArticle.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    <View>
+      <View key={article.id} className="article-title">
+        <Image
+          className={`article-image-${article.id}`}
+          source={{ uri: article.image }}
+          style={styles.image}
+        />
+        <Text id={"article-" + article.id + "-title"} style={styles.title}>
+          {article.title}
+        </Text>
```
Comment (erikbjoern) [inline src/components/SingleArticle.js]: We could be able to re-use the ArticleCard here

Comment (jysmys) [reply src/components/SingleArticle.js]: well the article card doesnt use the body of the article and it dont have to be touchable....

File: src/components/ArticleList.js (hunk 1 above)
Comment (jysmys) [reply src/components/ArticleList.js]: can be also

File: src/components/SingleArticle.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { View, Image, Text, StyleSheet } from "react-native";
+import { EBGaramond_400Regular } from "@expo-google-fonts/eb-garamond";
+
+const SingleArticle = ({ route }) => {
+  const { articleId } = route.params;
+  console.log(articleId);
```
Comment (ViaMarcus) [inline src/components/SingleArticle.js]: Is this intentional or leftover from debugging?

File: src/components/SingleArticle.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  };
+
+  useEffect(() => {
+    chooseArticle();
+  }, []);
+  const specificArticle = article && (
+    <View>
+      <View key={article.id} className="article-title">
+        <Image
+          className={`article-image-${article.id}`}
```
Comment (ViaMarcus) [inline src/components/SingleArticle.js]: You should try to be consistent between interpolation and concatenation (the ids below all use concatenation)

**Review by erikbjoern — APPROVED** [review]

File: src/components/SingleArticle.js (hunk 3 above)
Comment (jysmys) [reply src/components/SingleArticle.js]: leftover...

File: src/components/SingleArticle.js (hunk 4 above)
Comment (jysmys) [reply src/components/SingleArticle.js]: yes will try that

**Review by kermit-klein — APPROVED** [review]

File: src/components/ArticleCard.js [hunk 5]
```diff
@@ -1,35 +1,46 @@
...
+          articleId: props.article.id,
+        });
+      }}
+    >
+      <View>
+        <Image
+          source={{ uri: props.article.image }}
+          style={styles.image}
+        />
+        <Text id={`article-title-${props.article.id}`} style={styles.title}>
```
Comment (tochman) [inline src/components/ArticleCard.js:21]: Are you using this I’d for anything?

File: src/components/ArticleList.js (hunk 1 above)
Comment (tochman) [reply src/components/ArticleList.js]: I would go with an empty array as Erik suggested

File: src/components/ArticleList.js [hunk 6]
```diff
@@ -18,18 +17,17 @@ const ArticleList = () => {
...
-        data={articleCards}
-        renderItem={({ item }) => <View>{item.key}</View>}
-      />
-    </View>
+    <>
+      {articleList && (
+        <FlatList
+          data={articleList}
+          renderItem={({ item }) => (
+            <ArticleCard article={item} navigation={navigation} id={item.id+1}/>
```
Comment (tochman) [inline src/components/ArticleList.js]: Not sure about the id. Do you need it?

File: src/components/ArticleCard.js (hunk 5 above)
Comment (jysmys) [reply src/components/ArticleCard.js:21]: yes to find the element in the feature test

File: src/components/ArticleList.js (hunk 6 above)
Comment (jysmys) [reply src/components/ArticleList.js]: nope, i will remove it

File: src/components/ArticleList.js (hunk 1 above)
Comment (jysmys) [reply src/components/ArticleList.js]: changed!

**Review by ViaMarcus — APPROVED** [review]

**Pull request merged**